sdk/communication/communication-rooms/src/roomsClient.ts

```diff
@@ -149,127 +143,200 @@ export class RoomsClient {
+        roomId,
+        pageSettings.continuationToken,
+        options
+      );
+      pageSettings.continuationToken = currentSetResponse.nextLink;
+      if (currentSetResponse.value) {
+        yield currentSetResponse.value.map(mapToRoomParticipantSDKModel, this);
+      } else {
+        break;
+      }
+    }
+  }
+
+  private async *listParticipantsAll(
+    roomId: string,
+    options: ListParticipantsOptions = {}
+  ): AsyncIterableIterator<RoomParticipant> {
+    for await (const page of this.listParticipantsPage(roomId, {}, options)) {
+      yield* page;
+    }
   }
 
   /**
-   * Adds Participants to a room asynchronously
+   * Gets the participants of a room asynchronously.
    * @param roomId - ID of the room.
-   * @param request - Request for adding participants to a room.
    * @param options - Operational options.
    * @returns a list of all the participants in the room.
    */
-  public async addParticipants(
+  public listParticipants(
     roomId: string,
-    participants: RoomParticipant[],
-    options: AddParticipantsOptions = {}
-  ): Promise<void> {
-    return tracingClient.withSpan(
-      "RoomsClient-AddParticipants",
-      options,
-      async (updatedOptions) => {
-        await this.client.rooms.addParticipants(
-          roomId,
-          {
-            participants: participants.map((participant) =>
-              mapToRoomParticipantRestModel(participant)
-            ),
-          },
-          updatedOptions
-        );
-      }
+    options: ListParticipantsOptions = {}
+  ): PagedAsyncIterableIterator<RoomParticipant> {
+    const { span, updatedOptions } = tracingClient.startSpan(
+      "RoomsClient-GetParticipants",
+      options
     );
+    try {
+      const iter = this.listParticipantsAll(roomId, updatedOptions);
+      return {
+        next() {
+          return iter.next();
+        },
+        [Symbol.asyncIterator]() {
+          return this;
+        },
+        byPage: (settings: ListPageSettings = {}) => {
+          return this.listParticipantsPage(roomId, settings, updatedOptions);
+        },
+      };
+    } catch (e: any) {
+      span.setStatus({
+        error: e,
+        status: "error",
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
   }
 
   /**
    * Updates the Participants in a Room asynchronously.
+   * Participant is added to room if they did not exist and updated if already in room.
    * @param roomId - ID of the room.
-   * @param request - Request for updating participants in a room.
+   * @param participants - List of participants to upsert.
    * @param options - Operational options.
    * @returns a list of all the participants in the room.
    */
-  public async updateParticipants(
+  public async addOrUpdateParticipants(
     roomId: string,
-    participants: RoomParticipant[],
-    options: UpdateParticipantsOptions = {}
+    participants: RoomParticipantPatch[],
+    options: UpsertParticipantsOptions = {}
```

Comment:
Let's rename the option to AddOrUpdateParticipantsOptions. basically there should be no more reference to upsert in the repo anymore